File: accsaber-database/src/main/java/de/ixsen/accsaber/database/config/AccSaberSchemaFilter.java
package de.ixsen.accsaber.database.config;

import org.hibernate.boot.model.relational.Namespace;
import org.hibernate.boot.model.relational.Sequence;
import org.hibernate.mapping.Table;
import org.hibernate.tool.schema.spi.SchemaFilter;

public class AccSaberSchemaFilter implements SchemaFilter {
    public static final AccSaberSchemaFilter INSTANCE = new AccSaberSchemaFilter();

    @Override
    public boolean includeNamespace(Namespace namespace) {
        return true;
    }

    @Override
    public boolean includeTable(Table table) {
        return !table.getName().contains("acc_saber_score") && !table.getName().contains("acc_saber_player");
    }

    @Override
    public boolean includeSequence(Sequence sequence) {
        return true;
    }
}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/config/AccSaberSchemaFilter.kt
package de.ixsen.accsaber.database.config

import org.hibernate.boot.model.relational.Namespace
import org.hibernate.boot.model.relational.Sequence
import org.hibernate.mapping.Table
import org.hibernate.tool.schema.spi.SchemaFilter

class AccSaberSchemaFilter : SchemaFilter {
    override fun includeNamespace(namespace: Namespace): Boolean {
        return true
    }

    override fun includeTable(table: Table): Boolean {
        return !table.name.contains("acc_saber_score") && !table.name.contains("acc_saber_player")
    }

    override fun includeSequence(sequence: Sequence): Boolean {
        return true
    }

    companion object {
        val INSTANCE = AccSaberSchemaFilter()
    }
}